plugin/storage/integration/es_integration_test.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package integration
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/olivere/elastic"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+
+	"github.com/uber/jaeger/pkg/es"
+	"github.com/uber/jaeger/pkg/testutils"
+	"github.com/uber/jaeger/plugin/storage/es/spanstore"
+)
+
+const (
+	host          = "0.0.0.0"
+	queryPort     = "9200"
+	queryHostPort = host + ":" + queryPort
+	queryURL      = "http://" + queryHostPort
+	username      = "elastic"  // the elasticsearch default username
+	password      = "changeme" // the elasticsearch default password
+)
+
+func initializeES(s *StorageIntegration) error {
+	if s.reader != nil || s.writer != nil {
+		return nil
+	}
+	rawClient, err := elastic.NewClient(
+		elastic.SetURL(queryURL),
+		elastic.SetBasicAuth(username, password),
+		elastic.SetSniff(false))
+	if err != nil {
+		return err
+	}
+	client := es.WrapESClient(rawClient)
+	logger, _ := testutils.NewLogger()
+	writer := spanstore.NewSpanWriter(client, logger)
+	reader := spanstore.NewSpanReader(client, logger)
+
+	s.cleanUp = esCleanUp
+	s.refresh = esRefresh
+	s.cleanUp()
+	s.writer = writer
+	s.reader = reader
+	s.logger = logger
+	return nil
+}
+
+func esCleanUp() error {
+	simpleClient, err := elastic.NewSimpleClient(
```

Comment:
We're setting up a new es client every time this function gets called. I'd prefer if we create a ES StorageIntegration which embeds StorageIntegration. This way we can use all the functionality in StorageIntegration and initialize the es client just once.